packages/algob-web/src/types.ts

```diff
@@ -0,0 +1,281 @@
+import { Account as AccountSDK, LogicSig, LogicSigArgs } from 'algosdk';
+import * as z from 'zod';
+
+import type { ASADefSchema, ASADefsSchema } from "../../algob-web/src/types-input";
+
+export type AccountAddress = string;
+
+export interface AnyMap {
+  [key: string]: any // eslint-disable-line @typescript-eslint/no-explicit-any
+}
+
+/**
+ * After an asset has been created only the manager,
+ * reserve, freeze and reserve accounts can be changed.
+ * All other parameters are locked for the life of the asset.
+ */
+export interface AssetModFields {
+  manager?: string
+  reserve?: string
+  freeze?: string
+  clawback?: string
+}
+
+/**
+ * Common transaction parameters (fees, note..) */
+export interface TxParams {
+  /**
+   * feePerByte or totalFee is used to set the appropriate transaction fee parameter.
+   * If both are set then totalFee takes precedence.
+   * NOTE: SDK expects`fee: number` and boolean `flatFee`. But the API expects only one
+   * on parameter: `fee`. Here, we define feePerByte and totalFee - both as numberic
+   * parameters. We think that this is more explicit. */
+  feePerByte?: number
+  totalFee?: number
+  // The first round for when the transaction is valid.
+  firstValid?: number
+  // firstValid + validRounds will give us the ending round for which the transaction is valid.
+  validRounds?: number
+  // A lease enforces mutual exclusion of transactions.
+  lease?: Uint8Array
+  // Any data up to 1000 bytes.
+  note?: string
+  noteb64?: string
+  // When set, it indicates that the transaction is requesting
+  // that the Sender account should be closed, and all remaining
+  // funds, after the fee and amount are paid, be transferred to this address.
+  closeRemainderTo?: AccountAddress
+  // Specifies the authorized address.
+  rekeyTo?: AccountAddress
+  // you can learn more about these parameters here.(https://developer.algorand.org/docs/reference/transactions/#common-fields-header-and-type)
+}
+
+/**
+ * Stateful Smart contract flags for specifying sender and schema */
+export interface SSCDeploymentFlags extends SSCOptionalFlags {
+  sender: AccountSDK
+  localInts: number
+  localBytes: number
+  globalInts: number
+  globalBytes: number
+}
+
+/**
+ * Stateful smart contract transaction optional parameters (accounts, args..). */
+export interface SSCOptionalFlags {
```

Comment:
if we are changing this things, maybe we can consider rename `SSC` -> `App`. What do you think?

Reply:
but then we will have to change `deleteSSC` and `updateSSC` as well, that will be breaking changes, i think we can keep it same.

Reply:
I think `App` is more clear and more widely used  in the community than `SSC`.

Reply:
Ok, then will change all the ssc to app, and add in change log breaking change section